source/virtualBuffers/gecko_ia2.py

```diff
@@ -238,6 +264,35 @@ class Gecko_ia2(VirtualBuffer):
 	#: frame/iframe in the lists is a tuple of (IAccessible2_2, uniqueId). This
 	#: cache is used across instances.
 	_framesCache = weakref.WeakKeyDictionary()
+	_nativeAppSelectionMode = False
+
+	@script(
+		gesture="kb:NVDA+shift+f10"
+	)
+	def script_toggleNativeAppSelectionMode(self, gesture):

```

Comment:
this function also needs type information and translator comments on ui.messages